angular/src/providers/angular-delegate.ts

```diff
@@ -145,7 +147,23 @@ export const attachView = (
 
   const instance = componentRef.instance;
   const hostElement = componentRef.location.nativeElement;
+
   if (params) {
+    /**
+     * For modals and popovers, a reference to the component is
+     * added to `params` during the call to attachViewToDom. If
+     * a reference using this name is already set, this means
+     * the app is trying to use the name as a component prop,
+     * which will cause collisions.
+     */
+    if (elementReferenceKey && instance[elementReferenceKey] !== undefined) {
+      console.error(
+        `[Ionic Error]: ${elementReferenceKey} is a reserved property name when using ${container.tagName.toLowerCase()}. Check the component properties of ${
```

Comment:
~ Thoughts on reworking this message:

What do you think about this output?

> [Ionic Error]: "modal" is a reserved property of ion-modal. Rename or remove the "modal" property from CreateTransationComponent.

Reply:
Well, it's not necessarily a property _of_ `ion-modal` -- it actually winds up being a _reference_ to one -- so I worry that would make the needed action confusing, but I like the rest, much snappier 👀